test/core/security/grpc_tls_crl_provider_test.cc

```diff
@@ -67,21 +67,25 @@ namespace testing {
 class FakeDirectoryReader : public DirectoryReader {
  public:
   ~FakeDirectoryReader() override = default;
-  absl::StatusOr<std::vector<std::string>> GetDirectoryContents() override {
-    return files_in_directory_;
+  absl::Status ForEach(
+      absl::FunctionRef<void(absl::string_view)> callback) override {
+    if (!files_in_directory_.ok()) {
+      return files_in_directory_.status();
+    }
+    for (const auto& file : *files_in_directory_) {
+      callback(file);
+    }
+    return absl::OkStatus();
   }
-  absl::string_view Name() override { return kCrlDirectory; }
+  absl::string_view Name() const override { return kCrlDirectory; }
 
   void SetFilesInDirectory(absl::StatusOr<std::vector<std::string>> files) {
```

Comment:
Given that this interface requires the tests to declare an out-of-line vector, it might be better to split this into two methods:

```
void SetStatus(absl::Status status) { files_in_directory_ = std::move(status); }
void SetFilesInDirectory(std::vector<std::string> files) { files_in_directory_ = std::move(files); }
```

Then the tests can go back to specifying the vectors directly in the call to `SetFilesInDirectory()` instead of needing a local variable for the vector.

Reply:
I like it better this way too, it reduces cruft in the test to be able to do it inline in the fn call